hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java

```diff
@@ -179,6 +180,7 @@ public void rollback() {
 	public void commit() {
 		throwIfPendingStateIsInconsistent();
 		historian.addNewRecords();
+		historian.addNewEntities();
```

Comment:
In general I think this idea is really smart...any component can update the `TransactionContext` with expiring entities they've created; and then at the end of the transaction, one component takes care of tracking any such entities.